tools/query_breakdown/src/main/java/Node.java

```diff
@@ -0,0 +1,33 @@
+import java.util.ArrayList;
+
+/**
+ * This class represents a single node in the tree visualization of the algorithm. Below are fields
+ * necessary for implementation.
+ */
+public class Node {
+
+  // ensures tree structure
+  ArrayList<Node> children;
```

Comment:
Please use explicit access control, i.e. add "private"  for this member. Also do the same for other members.